File: Button/NotificationButton/NotificationButtonVBNET/NotificationButtonElement.vb
Imports System.ComponentModel
Imports Telerik.WinControls.UI
Imports Telerik.WinControls

Public Class NotificationButtonElement
Inherits RadItem
    Private buttonElement_Renamed As RadButtonElement
    Public Event Click As EventHandler
    Private result As Integer
    Private lve As LightVisualElement
    Private sizeConst As Integer

    Protected Overrides Sub CreateChildElements()
        MyBase.CreateChildElements()
        sizeConst= 20
        buttonElement_Renamed = New RadButtonElement()
        buttonElement_Renamed.Class = "RibbonBarButtonElement"
        buttonElement_Renamed.ButtonFillElement.Class = "ButtonInRibbonFill"
        buttonElement_Renamed.BorderElement.Class = "ButtonInRibbonBorder"
        buttonElement_Renamed.Margin = New System.Windows.Forms.Padding(0, 10, 10, 0)
        AddHandler buttonElement_Renamed.Click, AddressOf buttonElement_Click
        Me.Children.Add(buttonElement_Renamed)

        lve = New LightVisualElement()
        lve.BackColor = Color.Red
        lve.DrawFill = True
        lve.StretchHorizontally = False
        lve.StretchVertically = False
        lve.GradientStyle = Telerik.WinControls.GradientStyles.Solid
        lve.Shape = New RoundRectShape(sizeConst / 2)
        lve.SmoothingMode = System.Drawing.Drawing2D.SmoothingMode.AntiAlias
        lve.ForeColor = Color.White
        lve.MinSize = New Size(sizeConst, sizeConst)
        lve.Visibility = ElementVisibility.Collapsed
        Me.Children.Add(lve)
    End Sub

    Protected Overrides Sub OnPropertyChanged(ByVal e As RadPropertyChangedEventArgs)
        MyBase.OnPropertyChanged(e)

        If e.Property.Name = "Bounds" Then
            lve.Margin = New System.Windows.Forms.Padding(Me.Bounds.Width - sizeConst - 1, 0, 0, 0)
        End If
    End Sub

    Public Overrides Property Text() As String
        Get
            Return MyBase.Text
        End Get
        Set(value As String)
            MyBase.Text = value
            Me.ButtonElement.Text = value
        End Set
    End Property

    Private Sub buttonElement_Click(ByVal sender As Object, ByVal e As EventArgs)
        If Not ClickEvent Is Nothing Then
            RaiseEvent Click(Me.ButtonElement, e)
        End If
    End Sub

    Public ReadOnly Property ButtonElement() As RadButtonElement
        Get
            Return Me.buttonElement_Renamed
        End Get
    End Property

    <DefaultValue(0)> _
    Public Property NotificationCount() As Integer
        Get
            If Not lve.Text Is Nothing Then
                Dim isInt As Boolean = Integer.TryParse(lve.Text, result)
                If isInt Then
                    Return result
                End If
            End If

            Return 0
        End Get
        Set(value As Integer)
            If value > 0 Then
                lve.Visibility = ElementVisibility.Visible
            Else
                lve.Visibility = ElementVisibility.Collapsed
            End If
            lve.Text = value.ToString()
        End Set
    End Property
End Class

File: Button/NotificationButton/NotificationButtonCSharp/NotificationButtonElement.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Text;
using Telerik.WinControls;
using Telerik.WinControls.UI;

namespace NotificationButtonCSharp
{
    public class NotificationButtonElement : RadItem
    {
        RadButtonElement buttonElement;

        public event EventHandler Click;

        int result;
        LightVisualElement lve;
        int sizeConst = 20;

        protected override void CreateChildElements()
        {
            base.CreateChildElements();

            buttonElement = new RadButtonElement();
            buttonElement.Class = "RibbonBarButtonElement";
            buttonElement.ButtonFillElement.Class = "ButtonInRibbonFill";
            buttonElement.BorderElement.Class = "ButtonInRibbonBorder";
            buttonElement.Margin = new System.Windows.Forms.Padding(0, 10, 10, 0);
            buttonElement.Click += new EventHandler(buttonElement_Click);
            this.Children.Add(buttonElement);

            lve = new LightVisualElement();
            lve.BackColor = Color.Red;
            lve.DrawFill = true;
            lve.StretchHorizontally = false;
            lve.StretchVertically = false;
            lve.GradientStyle = Telerik.WinControls.GradientStyles.Solid;
            lve.Shape = new RoundRectShape(sizeConst / 2);
            lve.SmoothingMode = System.Drawing.Drawing2D.SmoothingMode.AntiAlias;
            lve.ForeColor = Color.White;
            lve.MinSize = new Size(sizeConst, sizeConst);
            lve.Visibility = ElementVisibility.Collapsed;
            this.Children.Add(lve);
        }

        protected override void OnPropertyChanged(RadPropertyChangedEventArgs e)
        {
            base.OnPropertyChanged(e);

            if (e.Property.Name == "Bounds")
            {
                lve.Margin = new System.Windows.Forms.Padding(this.Bounds.Width - sizeConst - 1, 0, 0, 0);
            }
        }

        public override string Text
        {
            get
            {
                return base.Text;
            }
            set
            {
                base.Text = value;
                this.ButtonElement.Text = value;
            }
        }

        void buttonElement_Click(object sender, EventArgs e)
        {
            if (Click != null)
                Click(this.ButtonElement, e);
        }

        public RadButtonElement ButtonElement
        {
            get
            {
                return this.buttonElement;
            }
        }

        [DefaultValue(0)]
        public int NotificationCount
        {
            get
            {
                if (lve.Text != null)
                {
                    bool isInt = int.TryParse(lve.Text, out result);
                    if (isInt)
                    {
                        return result;
                    }
                }

                return 0;
            }
            set
            {
                if (value > 0)
                {
                    lve.Visibility = ElementVisibility.Visible;
                }
                else
                {
                    lve.Visibility = ElementVisibility.Collapsed;
                }
                lve.Text = value.ToString();
            }
        }
    }
}